compiler/rustc_attr/src/builtin.rs

```diff
@@ -161,6 +151,20 @@ pub enum StabilityLevel {
     },
 }
 
+/// The available const-stability levels for const functions.
+/// For details see [#131349](https://github.com/rust-lang/rust/pull/131349).
+#[derive(Encodable, Decodable, PartialEq, Clone, Debug, Eq, Hash)]
+#[derive(HashStable_Generic)]
+pub enum ConstStabilityLevel {
+    /// For functions declared const-stable
+    Stable { since: StableSince },
+    /// For functions declared const-unstable
+    Unstable { unstables: SmallVec<[Unstability; 1]> },
+    /// For functions with no explicit const-stability attribute that require checking recursive
+    /// const stability. This is either an unmarked const fn or a `const_stable_indirect` intrinsic.
+    Implicit,
```

Comment:
Please rename this to `ImplicitUnstable`, because we never want anything to be "implicitly stable".

Also FWIW this variant will disappear when I have the time to continue working on the const stability checks.

Reply:
Done. Do you think this should be blocked on that being merged?

Reply:
My PR is now up: https://github.com/rust-lang/rust/pull/132541